cogs/Bartender.py

```diff
@@ -1,25 +1,70 @@
-    FACTORY_TYPE = 'drink'
+BARTENDER_DESCRIPTION = 'Used to provide any bud with a delicious, sensual drink.'
+DRINK_DESCRIPTION = 'Get a nice, refreshing drink from the Bartender'
+DRINK_USAGE = '--> Gets a random drink\n.drink a --> Gets an alcoholic drink\n.drink na --> gets a nonalcoholic drink\n.drink [bourbon,brandy,cognac,gin,rum,scotch,tequila,vodka,whiskey,wine] --> Gets an alcoholic beverage of the given type'
+class Bartender(commands.Cog, name='Bartender', description=BARTENDER_DESCRIPTION):
     def __init__(self, bot):
         self.bot = bot
     
-    @commands.command(name='drink', description='Get a nice, refreshing drink from the Bartender',aliases=['drinks', 'cocktail', 'cocktails'])
-    async def drink(self, ctx):
-        DRINK_URL = 'https://www.thecocktaildb.com/api/json/v1/1/random.php'
+    @commands.command(name='drink', description=DRINK_DESCRIPTION, usage=DRINK_USAGE)
+    async def drink(self, ctx, *args):
+        FILTER_TYPES = ['a','bourbon','brandy','cognac','gin', 'na','rum','scotch','tequila','vodka','whiskey','wine']
+        drink_url = 'https://www.thecocktaildb.com/api/json/v1/1/'
+
         try:
-            json = req.get(DRINK_URL).json()['drinks'][0]
-            drink = Drink(json)
-            await ctx.send(embed = drink.embed)
+            if not args:
+                drink_url += 'random.php'
+                json = req.get(drink_url).json()
+                drink = Drink(json['drinks'][0])
+                await ctx.send(embed = drink.embed)
+                return
+            
+
+            if len(args) > 1:
+                await ctx.send(f'I think you added too many arguments there, bud.')
+                return
+
+            arg = args[0].lower()
+            if arg not in FILTER_TYPES:
+                    message = 'Better check your arguments there, bud.'
+                    chance = r.randint(1,100)
+                    if chance > 80:
+                        message += ' (Kinda cringe...)'
+                    await ctx.send(message)
+                    return
+            
+            
+            if arg == 'a':
+                filter = 'filter.php?a=Alcoholic' 
+                resolved_url = self.drink_url_factorty(drink_url, filter)
+            elif arg == 'na':
+                filter = 'filter.php?a=Non_Alcoholic'
+                resolved_url = self.drink_url_factorty(drink_url, filter)
+            else:
+                filter = f'filter.php?i={arg}'
+                resolved_url = self.drink_url_factorty(drink_url, filter)
             
+            json = req.get(resolved_url).json()
+            drink = Drink(json['drinks'][0])
+            await ctx.send(embed = drink.embed)
+
         except Exception as ex:
             traceback.print_exc()
             await ctx.send(f'Ayo, your code is wack.\n Error: {ex}')
-    
+
+
+    def drink_url_factorty(self, drink_url, filter):
```

Comment:
1. "factory" is spelled incorrectly
2. This isn't really a 'factory' so to speak (as in the factory pattern). The factory pattern involves creating an instance of a polymorphic type at run-time. This is kind of just a normal function that gets a random drink given some filter. 
3. Continuing off the last point, the actual behavior of this function isn't described by its name. You aren't just constructing a URL from the given filter, you're finding the URL for a random drink given this filter. Something like `random_drink_url_from_filter(...)` is more appropriate (though admittedly more verbose - you could maybe include the next step in the process where you construct a drink object from this URL and call it `random_drink_from_filter(...)`, or make that part a helper function and include it inside this function).

Reply:
Refactored to reduce confusion:
https://github.com/DonChalupa/Robbot/pull/16/commits/2b02393314efb1d65289e7c5c697dcbffcb6379d#diff-a1132eee6c43b832e1b4945ce331759b2cdb8e84dbe814c5cc4e4a31020ec7b1R63